File: fcp/artifact_building/test_utils.py
# Copyright 2022 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#      http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
"""Utilities used in tests."""

import tensorflow as tf

from fcp.protos import plan_pb2


def set_checkpoint_op(
    checkpoint_op_proto: plan_pb2.CheckpointOp,
    saver: tf.compat.v1.train.SaverDef,
):
  """Sets the saver_def from saver onto checkpoint_op_proto and fixes a name."""
  if not saver:
    return
  saver_def_proto = checkpoint_op_proto.saver_def

  saver_def_proto.CopyFrom(saver.as_saver_def())
  # They are calling an Op a Tensor and it works in python and
  # breaks in C++.  However, for use in the python Saver class, we
  # need the tensor because we need sess.run() to return the
  # tensor's value. So, we only strip the ":0" in the case of
  # plan execution, where we use the write_checkpoint and
  # read_checkpoint methods below instead of the Saver.
  saver_def_proto.save_tensor_name = saver_def_proto.save_tensor_name.replace(
      ':0', ''
  )
  assert saver_def_proto.save_tensor_name.rfind(':') == -1

File: fcp/confidentialcompute/test_utils.cc
/*
 * Copyright 2025 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

#include "fcp/confidentialcompute/test_utils.h"

#include <optional>
#include <string>

#include "google/protobuf/any.pb.h"
#include "fcp/protos/confidentialcompute/file_info.pb.h"
#include "federated_language/proto/computation.pb.h"
#include "tensorflow_federated/proto/v0/executor.pb.h"

namespace fcp {
namespace confidential_compute {

tensorflow_federated::v0::Value CreateFileInfoValue(
    const std::string& uri, std::optional<std::string> key,
    std::optional<federated_language::Type> type) {
  fcp::confidentialcompute::FileInfo file_info;
  file_info.set_uri(uri);
  if (key.has_value()) {
    file_info.set_key(*key);
  }
  tensorflow_federated::v0::Value value;
  value.mutable_computation()->mutable_data()->mutable_content()->PackFrom(
      file_info);
  if (type.has_value()) {
    *value.mutable_computation()->mutable_type() = *type;
  }
  return value;
}

}  // namespace confidential_compute
}  // namespace fcp
